### PR TITLE
SS-28758: Integrate RdKitJS library for certain server functionalities

### PR DESCRIPTION
**Pull request by apar03**
- Added a simple plug-n-play architecture, for adding new functionalities that utilize methods from the [RDKitJS repo](https://github.com/rdkit/rdkit/tree/master/Code/MinimalLib)
- Migrated 4 of the server side functionalities to utilize the corresponding RDKit JS lib methods: 
(i) Aromatize: `get_aromatic_form`
(ii) Dearomatize: `get_kekule_form`
(iii) 2D Clean: `get_new_coords`
(iv) Calculate CIP : `get_stereo_tags`

- Downloaded the already built JS & Wasm files from [here](https://rdkit.org/temp/demo/)
- Verified that these functionalities work fine, could be tried out at http://ld-desis.dev.bb.schrodinger.com:9145/livedesign/ketcher2/ketcher.html
- Added 'how-to-add' RDkit JS lib functionality steps in the existing document maintained [here](https://docs.google.com/document/d/1mPmgPP26_Mi4swJamIYLBWoqUCJ2bNeiGbwbbVpDmEk/edit)

Primary - @mycrobe

### CODE REVIEW

**Comment by apar03** [comment]
Please note that this is a fresh PR, which has been migrated from the earlier private repo (which we're going to delete).
This PR contains the latest code state, so please consider this as the source of truth for further comments and sign-off.

cc @wcjordan @mycrobe

File: src/script/ui/state/rdkitjs/index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+			fieldName: 'CIP_DESC',
+			fieldValue: '(R)',
+			init: true
+		};
+		const action = new Action();
+		deleteAllSGroupsWithName(restruct, action, attributes.fieldName);
+		result.CIP_atoms.forEach((a) => {
+			const [index, fieldValue] = a;
+			const atomIdx = keys[index];
+			const atom = restruct.molecule.atoms.get(atomIdx);
```
Comment (mycrobe) [inline src/script/ui/state/rdkitjs/index.js]: can you briefly explain why there are two array indices here to find an atom?

Comment (apar03) [reply src/script/ui/state/rdkitjs/index.js]: Honestly, I have little knowledge about this method.. (also `deleteAllSGroupsWithName` below)

I've simply copied over these methods as-is from Nic's commit here - https://github.com/ZontaNicola/ketcher/commit/2fb10a08df4ebc57d34af1cbd481bca882761580

Nic had hooked up the 'Calculate CIP' functionality with RDKit using an API call to the flask server. I have extracted these helper methods, and have integrated the functionality with RDKit JS.

@ZontaNicola do you know an answer to Joe's question?

File: src/script/ui/state/rdkitjs/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			context: 'Atom',
+			fieldName: 'CIP_DESC',
+			fieldValue: '(R)',
+			init: true
+		};
+		const action = new Action();
+		deleteAllSGroupsWithName(restruct, action, attributes.fieldName);
+		result.CIP_atoms.forEach((a) => {
+			const [index, fieldValue] = a;
+			const atomIdx = keys[index];
```
Comment (mycrobe) [inline src/script/ui/state/rdkitjs/index.js]: Could you rename this to `atomKey`?

Is a `key` a string?

**Review by mycrobe — CHANGES_REQUESTED** [review]
I think these functions need some tests so we can verify that they do what we expect them to do. If there are plans to introduce those in another PR, then this could be merged.

The code seems fine to me, but I'm not familiar with RDKit.

**Comment by apar03** [comment]
> If there are plans to introduce those in another PR, then this could be merged.

Yeah, that is the plan. Thanks for pointing!

> The code seems fine to me, but I'm not familiar with RDKit.

Yeah, @greglandrum had previously given an LGTM on the RDKit side of things https://github.com/schrodinger/ketcher-schrodinger/pull/3#pullrequestreview-359060396

File: src/script/ui/state/rdkitjs/index.js (hunk 2 above)
Comment (apar03) [reply src/script/ui/state/rdkitjs/index.js]: Sure, renamed.

Yes it's a string

**Comment by apar03** [comment]
Merging in the code changes following Joe's approval above.

**Pull request merged**